src/routes/tools/local-climate-change-snapshot/_data.js

```diff
@@ -200,3 +221,60 @@ export function getQueryParams({ location, boundary, imperial = true, stat }) {
   }
   return { params, method };
 }
+
+/**
+ * Calculate 30 year average for each scenario & period combination
+ * @param {array} scenarios
+ * @param {array} periods
+ * @return {array} scenario & period combinations that have a valid 30 year avg value
+ */
+export function calc30yAvgByPeriod(scenarios, periods = DEFAULT_STAT_PERIODS) {
+  const data = scenarios.map((d) => {
+    const { values, id: scenarioId, label: scenarioLabel } = d;
+    const dataByPeriods = periods.map((period) => {
+      const { id: periodId, label: periodLabel, start, end } = period;
+      const filteredValues = values.filter(
+        ({ date }) =>
+          date.getUTCFullYear() >= start && date.getUTCFullYear() <= end
+      );
+      let avg = null;
+      if (filteredValues.length) {
+        avg = mean(filteredValues, (d) => d.value);
+      }
+      return { periodId, periodLabel, scenarioId, scenarioLabel, avg };
+    });
+    return [...dataByPeriods];
+  });
+  return merge(data).filter(({ avg }) => avg);
+}
+
+/**
+ * Map 30 year extent for each scenario & period combination
+ * @param {array} scenarios
+ * @param {array} periods
+ * @return {array} scenario & period combination that have a valid 30 year extent
+ */
+export function map30yExtentByPeriod(
+  scenarios,
+  periods = DEFAULT_STAT_PERIODS
+) {
+  const data = scenarios.map((d) => {
+    const { values, id: scenarioId, label: scenarioLabel } = d;
+    const dataByPeriods = periods.map((period) => {
+      const { id: periodId, label: periodLabel, start, end } = period;
+      const filteredValues = values.filter(
+        ({ date }) =>
+          date.getUTCFullYear() >= start && date.getUTCFullYear() <= end
+      );
+      let min = null;
+      let max = null;
+      if (filteredValues.length && filteredValues.length === 1) {
+        min = filteredValues[0].min;
+        max = filteredValues[0].max;
+      }
+      return { periodId, periodLabel, scenarioId, scenarioLabel, min, max };
+    });
+    return [...dataByPeriods];
+  });
+  return merge(data).filter(({ min, max }) => min && max);
```

Comment:
if `min` or `max` is ever `0` here then those values will be filtered out, is that intended?

Reply:
Good catch. `0` is a valid value. Added a strict check for `null` in https://github.com/berkeley-gif/caladapt-website-2021/pull/209/commits/a0819d5619e39c5361059fc4870a8fe185e4b9fc